internal/storage/parser.go

```diff
@@ -22,11 +22,19 @@ func (e UnknownParserErr) Error() string {
 // Parser can be used to load and save files from/to disk.
 type Parser interface {
 	// FromBytes returns some Data that is represented by the given bytes.
-	FromBytes(byteData []byte) (interface{}, error)
+	FromBytes(byteData []byte) (RealValue, error)
```

Comment:
I'm hesitant to force all parsers to return a `RealValue`.

This was originally only meant to allow certain parsers to be more flexible with values when they need additional context in `ToBytes`, e.g. knowing that it is a multi-document slice.

I'm not sure I see the benefit of all parsers returning this since it adds additional complexity to the more basic implementations.

Additionally `FromBytes` and `ToBytes` now deal with different value types which I think is a little confusing.